modules/afmongodb/afmongodb-worker.c

```diff
@@ -29,15 +29,79 @@
 #include "value-pairs/value-pairs.h"
 #include "scanner/list-scanner/list-scanner.h"
 
+static LogThreadedResult _do_bulk_flush(MongoDBDestWorker *self);
+
+static void
+_compose_bulk_op_options(MongoDBDestWorker *self)
+{
+  MongoDBDestDriver *owner = (MongoDBDestDriver *) self->super.owner;
+
+  if (owner->use_bulk)
+    {
+      self->bson_opts = g_new0(bson_t, 1);
```

Comment:
Optional:
`bson_new()` and `bson_destroy()` may be easier as we could avoid the ` bson_destroy(self->bson_opts); g_free(self->bson_opts);` 2 step free.